diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32d7877..9fac229 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,6 +346,9 @@ if (CMAKE_SYSTEM_NAME STREQUAL "iOS")
         COMMAND ${CMAKE_COMMAND} -E copy_if_different
             "${CMAKE_SOURCE_DIR}/recompcontrollerdb.txt"
             "$<TARGET_BUNDLE_DIR:BanjoRecompiled>/recompcontrollerdb.txt"
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            "${CMAKE_SOURCE_DIR}/icons/app.png"
+            "$<TARGET_BUNDLE_DIR:BanjoRecompiled>/BanjoPadIcon.png"
         COMMENT "Copying BanjoPad iOS resources"
     )
 elseif (APPLE)
