diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7fdec9..17a847a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,14 +91,7 @@ else()
     add_subdirectory(src/tools/file_to_c)
 endif()
 add_subdirectory(src/contrib/re-spirv)
-if(CMAKE_SYSTEM_NAME STREQUAL "iOS")
-    add_library(nfd INTERFACE)
-    target_include_directories(nfd INTERFACE
-        "${PROJECT_SOURCE_DIR}/src/contrib/nativefiledialog-extended/src/include"
-    )
-else()
-    add_subdirectory(src/contrib/nativefiledialog-extended)
-endif()
+add_subdirectory(src/contrib/nativefiledialog-extended)
 add_subdirectory(src/contrib/zstd/build/cmake)
 
 if (CMAKE_SYSTEM_NAME MATCHES "Linux" AND RT64_SDL_WINDOW_VULKAN)
diff --git a/src/apple/rt64_apple.h b/src/apple/rt64_apple.h
index e8d49bf..359ac5f 100644
--- a/src/apple/rt64_apple.h
+++ b/src/apple/rt64_apple.h
@@ -5,3 +5,4 @@
 #pragma once
 
 const char* GetHomeDirectory();
+const char* GetApplicationSupportDirectory();
diff --git a/src/apple/rt64_apple.mm b/src/apple/rt64_apple.mm
index 4926c4d..a7eeaf5 100644
--- a/src/apple/rt64_apple.mm
+++ b/src/apple/rt64_apple.mm
@@ -5,3 +5,9 @@
 const char* GetHomeDirectory() {
     return strdup([NSHomeDirectory() UTF8String]);
 }
+
+const char* GetApplicationSupportDirectory() {
+    NSArray<NSURL *> *urls = [[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory inDomains:NSUserDomainMask];
+    NSURL *url = [urls firstObject];
+    return url == nil ? nullptr : strdup([[url path] UTF8String]);
+}
diff --git a/src/common/rt64_user_paths.cpp b/src/common/rt64_user_paths.cpp
index f38d8eb..342f35b 100644
--- a/src/common/rt64_user_paths.cpp
+++ b/src/common/rt64_user_paths.cpp
@@ -10,6 +10,7 @@
 #elif defined(_WIN32)
 #   include <Shlobj.h>
 #elif defined(__APPLE__)
+#   include <TargetConditionals.h>
 #   include "apple/rt64_apple.h"
 #endif
 
@@ -32,6 +33,11 @@ namespace RT64 {
         }
 
         CoTaskMemFree(knownPath);
+#   elif defined(__APPLE__) && TARGET_OS_IPHONE
+        const char *applicationSupportDir = GetApplicationSupportDirectory();
+        if (applicationSupportDir != nullptr) {
+            resultPath = std::filesystem::path{ applicationSupportDir } / appId;
+        }
 #   elif defined(__linux__) || defined(__APPLE__)
         const char *homeDir = getenv("HOME");
         if (homeDir == nullptr) {
